cleanup_delete.go

```diff
@@ -152,5 +124,29 @@ func (dq *deleteQueue) start(ctx context.Context) {
 				dq.deleteFromPending()
 			}
 		}
+		select {
+		case <-ctx.Done():
+			dq.drain()
+			// drain the delete queue and return
+			go func() {
+				for {
```

Comment:
It isn't clear why this gofunc to grain is needed right after you call a message to drain the queue. Doesn't this gofunc leak as it never exits?